core/src/main/java/org/mule/api/util/CredentialsMaskUtil.java

```diff
@@ -79,22 +84,38 @@ public static String maskUrlUserAndPassword(String input, Pattern passwordPatter
     }
 
     /**
-     * masks password attribute
+     * Masks password attribute
      * 
      * @param password password to be masked
-     * 
      * @return password masked
      */
     public static String maskPasswordAttribute(String password)
     {
         return format(PASSWORD_ATTRIBUTE_MASK, password);
     }
 
-    private static String maskUrlPattern(String input, Pattern pattern, String mask) {
+    /**
+     * Masks pattern in input
+     * 
+     * @param input credentials for user and password to be masked
+     * @param passwordPattern password pattern
+     * @param mask mask to use
+     * @return credentials masked
+     */
+    public static String maskUrlPattern(String input, Pattern pattern, String mask)
+    {
         return maskUrlPattern(input, pattern, mask, "");
     }
-    
-    private static String maskUrlPattern(String input, Pattern pattern, String mask, String prefix) {
+
+    /**
+     * @param input credentials for user and password to be masked
+     * @param passwordPattern password pattern
+     * @param mask mask to use
+     * @param prefix prefix in credentials
+     * @return credentials masked
+     */
+    private static String maskUrlPattern(String input, Pattern pattern, String mask, String prefix)
```

Comment:
There is no need to maintain this javadoc as the method is private. If you want to keep it, apply the same fixes as requested in the previous comment . Also add method description